### PR TITLE
Fix unexpected behavior in UG due to invalid HTML markup

### LINKED ISSUE #1794: Unexpected siteNav scrolling effect on small screens
**Tell us about your environment**
Seen on markbind.org
* **MarkBind Version:**
v3.1.1

**What did you do? Please include the actual source code causing the issue.**
Reproduce: 
1. go to https://markbind.org/userGuide/formattingContents.html
2. reduce the width of the browser until siteNav collapse to an icon
3. click to open siteNav
4. scroll downwards

Some sort of delay/duplicate effect when scrolling.
- Suspected to be caused by the existence of pageNav, but that doesn't seem to be the case
- So far only the above page seems to be problematic

**What did you expect to happen?**
It should behave in the same manner as other pages, e.g. https://markbind.org/userGuide/authoringContents.html

**What actually happened? Please include the actual, raw output.**
hiDbUAYibU

### LINKED ISSUE #1840: Typo in UG: Syntax Cheat Sheet
**Tell us about your environment**
Windows
* **MarkBind Version:**
v3.1.1
**What did you do? Please include the actual source code causing the issue.**
Found two syntax examples with an issue, in [Syntax Cheat Sheet](https://markbind.org/userGuide/syntaxCheatSheet.html) page
**What did you expect to happen?**
No typo/incorrect examples

**What actually happened? Please include the actual, raw output.**
image
image

### PR DESCRIPTION
**Pull request by tlylt**
**What is the purpose of this pull request?**

- [x] Documentation update
- [x] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] Others, please explain:

**Overview of changes:**
Fixes #1794
Also fixes #1840 (very much related so putting it together here)

- Include a 'troubleshooting' section in UG for putting such errors

**Anything you'd like to highlight / discuss:**
Not too sure why, but sometimes it's fine for block-level elements to be in `<span>` but sometimes not.
E.g. in [embed](https://markbind.org/userGuide/formattingContents.html#embeds) section, seems like this span tag works fine and does not trigger any vue warning.
image

**Testing instructions:**
- verify the behavior according to issue #1794
- preview the [added section](https://deploy-preview-1837--markbind-master.netlify.app/userguide/troubleshooting)

**Proposed commit message: (wrap lines at 72 characters)**
Fix unexpected behavior in UG due to invalid HTML markup

Invalid HTML markup may cause Vue hydration issues.

Let's replace span tags with div tags to avoid invalid
inclusion of block-level elements in span tags.

This prevents possible errors in HTML syntax.

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes

### CODE REVIEW

File: docs/userGuide/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Bees   |     no     |    20 |
+Cats   |    yes     |   100 |
+</span>
+```
+
+The table specified by the markdown syntax above will be rendered as a block-level element, which will be included in a inline span element. This makes the HTML output invalid. A possible fix for the above situation is to wrap the table in a `<div>` element instead.
+
+<panel header="Example Underlying Error" type="seamless">
+
+vue.js:634 [Vue warn]: The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside `<p>`, or missing `<tbody>`. Bailing hydration and performing full client-side render.
```
Comment (ryoarmanda) [inline docs/userGuide/troubleshooting.md:37]: We can format the example error a little bit nicer, maybe a fenced code block if possible?

**Review by ryoarmanda — COMMENTED** [review]
Interesting issue 👀 I'm also a bit intrigued on the embed case, maybe the issue is for specific block-level elements like tables? (cc @jonahtanjz) I think we can also change those to use div for posterity sake, what do you think?

The docs you have written up are a good start, maybe some little stylization nits below:

**Comment by jonahtanjz** [comment]
> I'm also a bit intrigued on the embed case, maybe the issue is for specific block-level elements like tables? (cc @jonahtanjz)

Having block elements within inline elements should not be allowed according to the [SSR guide for Vue](https://vuejs.org/guide/scaling-up/ssr.html#hydration-mismatch). For some reason, Vue does not always warn or have issues with some of the cases.

For example, putting the following code on a MarkBind site,
```.html
<span>
<div>Test</div>
</span>
```
doesn't seem to have any warnings.

Similarly, even using the example provided on [Vue's page](https://vuejs.org/guide/scaling-up/ssr.html#hydration-mismatch),
```.html
<p>
<div>Test</div>
</p>
```
no warnings as well.

Putting a table in a `<span>` seems to consistently cause the warning to show up.

The following seems to cause the warning to show up as well,
```.html
<p>
<ul>
<li>1</li>
<li>2</li>
</ul>
</p>
```


> I think we can also change those to use div for posterity sake, what do you think?

Although some of the offending components don't result in any warnings, I think they should still be changed as well since they are not valid HTML and may cause some problematic issues in the future.

**Review by jonahtanjz — COMMENTED** [review]
Thanks @tlylt! 

A couple more invalid HTML:

* [ ] [docs/index.md#19](https://github.com/MarkBind/markbind/blob/master/docs/index.md?plain=1#L9) (`<p>` in `<span>`)
* [ ] [docs/userGuide/addingPages.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/addingPages.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/deployingTheSite.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/deployingTheSite.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/formattingContents.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/formattingContents.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/makingTheSiteSearchable.md#19](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/makingTheSiteSearchable.md?plain=1#L19) (`<p>` in `<span>`)
* [ ] [docs/userGuide/markBindInTheProjectWorkflow.md#16](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/markBindInTheProjectWorkflow.md?plain=1#L16) (`<p>` in `<span>`)
* [ ] [docs/userGuide/markBindSyntaxOverview.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/markBindSyntaxOverview.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/reusingContents.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/reusingContents.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/siteJsonFile.md#10](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/siteJsonFile.md?plain=1#L10) (`<p>` in `<span>`)
* [ ] [docs/userGuide/themes.md#16](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/themes.md?plain=1#L16) (`<p>` in `<span>`)
* [ ] [docs/userGuide/usingHtmlJavaScriptCss.md#34](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/usingHtmlJavaScriptCss.md?plain=1#L34) (example used is not valid HTML as it will create a `<p>` in `<span>`)
* [ ] [docs/userGuide/syntax/tree.md#96 and #110](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/syntax/tree.md?plain=1#L96) (codeblock and `<tree>` in `<span>`)
* [ ] [docs/userGuide/syntax/popovers.md#48](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/syntax/popovers.md?plain=1#L48) (`<div>` in `<p>`)

Also noticed a few of our Vue components having block-level elements in `<span>`. This can be fixed in another PR.

[Popover.vue](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/Popover.vue#L2)
[MinimalPanel.vue](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/panels/MinimalPanel.vue#L2)
[NestedPanel.vue](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/panels/NestedPanel.vue#L2)

**Comment by tlylt** [comment]
> A couple more invalid HTML:

Good catch! I have fixed the cases mentioned. I also:
- added [one warning](https://deploy-preview-1837--markbind-master.netlify.app/userguide/reusingcontents#includes) in the reusing content -> includes section to inform users to choose div over span
- removed some span to div replacement as those spans within a boilerplate appear to be important for the variable to work. Will open a separate issue for this.

**Review by jonahtanjz — APPROVED** [review]
Thanks @tlylt! Looks good to me 👍 

Let's wait for @ryoarmanda to look through and approve as well :)

**Review by ryoarmanda — APPROVED** [review]
Looks good to me as well! Will merge tonight

**Comment by ryoarmanda** [comment]
Some weird thing happening here 🤔 the checks have been in queue for a while now, and I didn't see it ever running. Even the checks tab show 0 checks, and now workflow records appear.

If this keeps going, I might have to close and reopen this PR to kick the checks back up again.

**Comment by tlylt** [comment]
> Some weird thing happening here 🤔 the checks have been in queue for a while now, and I didn't see it ever running. Even the checks tab show 0 checks, and no workflow records appear.
> 
> 
> 
> If this keeps going, I might have to close and reopen this PR to kick the checks back up again.

GitHub is experiencing some operational error, we can probably wait out.

**Comment by jonahtanjz** [comment]
> If this keeps going, I might have to close and reopen this PR to kick the checks back up again.

Will need to try this since it is still stuck in queue.

**Comment by ryoarmanda** [comment]
Okay, checks are running now.

**Pull request merged**